fms/modules/embedding.py

```diff
@@ -1,21 +1,129 @@
-import math
-from typing import List, Tuple, Union
+from typing import Optional
 
 import torch
 import torch.nn as nn
-from numpy import sign
 from torch.distributed.distributed_c10d import ProcessGroup
 
 from fms.distributed.tensorparallel import (
     all_gather_from_tensor_model_parallel_region,
     apply_colwise_tp,
     apply_embedding_tp,
-    apply_rowwise_tp,
     copy_to_tensor_model_parallel_region,
-    reduce_from_tensor_model_parallel_region,
 )
 
 
+class AbsolutePositionEmbedding(nn.Module):
```

Comment:
Is the intent to remove this functionality from WordEmbedding?

I do favor smaller PRs but maybe those changes should be together to clarify the refactoring, if that's the intent.

Reply:
Yes, that is a good point, I didn't want to change that class, but it makes sense to do that now. I will make that change.

Reply:
Removed abs pos logic from WordEmbedding and TP variant as only model using it is LLaMA and it has it set to false